src/v/cluster/node/types.h

```diff
@@ -56,12 +56,57 @@ struct local_state
         }
     }
 
+    // The non-const version above is used by some internal routines that
+    // manipulate state. This is just for reading.
+    const storage::disk& get_cache_disk() const {
+        if (cache_disk.has_value()) {
+            return cache_disk.value();
+        } else {
+            return data_disk;
+        }
+    }
+
     bool shared_disk() const { return !cache_disk.has_value(); }
 
     /// Report a generalized node-wide disk alert state: this is the worst of
     /// all drive's alert state.
     storage::disk_space_alert get_disk_alert() const;
 
+    /*
+     * the system may try to use less than the entire disk for log data. for
+     * example a configuration may be:
+     *
+     *    physical disk size: 1 TB
+     *
+     *    // configured reservations
+     *    target_size: 700 GB
+     *    cache size: 200 GB
+     *    overhead size: 100 GB
+     *
+     *    // current usage
+     *    current_size: 700 GB (at capacity)
+     *    reclaimable_size: 200 GB
+     *
+     * in this scenario scenario even though the target size has been reached,
+     * according to the retention policy on the node up to 200 GB is currently
+     * reclaimable if the target size is exceeded.
+     *
+     * data_target_size: the target capacity of log data
+     * data_current_size: current amount of log data
+     * data_reclaimable_size: amount of data reclaimable log data
+     */
+    struct log_data_state
+      : serde::envelope<
+          log_data_state,
+          serde::version<0>,
+          serde::compat_version<0>> {
+        uint64_t data_target_size{0};
+        uint64_t data_current_size{0};
+        uint64_t data_reclaimable_size{0};
+        friend std::ostream& operator<<(std::ostream&, const log_data_state&);
+    };
+    tristate<log_data_state> log_data_size{std::nullopt};
```

Comment:
nit: why does this have to be a tristate? From the pov of the caller, the data is either available or not.

Reply:
Ahh, right. I will add a comment to explain this. We have three cases that we wanted to report to the balancer:

1. early bootup--health report produced, but space manager hasn't yet had the chance to figure out what's current state is.
2. configuration on the node has disabled the feature entirely
3. real results being reported

The distinction with (1) is that the balancer will wait until it gets more information before acting on state in order to avoid the case where we might report a large disk and usage information and then later when space manager starts changing that state is a drastic way.

Reply:
Update. After speaking with Bharath, we decided to go with this simpler interpretation of available or not available which also allows us to remove the feature flag. I think it's much simpler now to reason through. Thanks for point that out.